connect_ext_test.go

```diff
@@ -541,6 +542,38 @@ func TestCompressMinBytes(t *testing.T) {
 	})
 }
 
+func Test_CustomCompression(t *testing.T) {
+	t.Parallel()
+	mux := http.NewServeMux()
+	compressionName := "deflate"
+	decompressor := func() connect.Decompressor {
+		return newDeflateReader(http.NoBody)
+	}
+	compressor := func() connect.Compressor {
+		w, err := flate.NewWriter(&strings.Builder{}, flate.DefaultCompression)
+		if err != nil {
+			t.Fatalf("failed to create flate writer: %v", err)
```

Comment:
It might be nice if the interface supported returning an error here, because we could fail back to identity encoding, etc. if the compression has issues. Right now, the only fallback for real use cases would be using `panic`.

Reply:
Yeah, since the writer constructor returns an error, it would be nice to handle that. On the other hand, the writer could be constructed outside of the `compressor` but then it would be leaking outside of the `compressor`.

Reply:
This is an interesting point, because on one hand, the writer could be constructed outside of the closure, however, it does make sense to encapsulate the writer type inside the closure and not have it leak.